[BE] Cleanup old ExecuTorch codegen and runtime code

larryliu0820 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #154165

Summary: These files are added to pytorch/pytorch before ExecuTorch is
opensourced. Now is a good time to remove it from pytorch/pytorch, since
the code is moved to pytorch/executorch already.

Test Plan: Rely on CI jobs.

cc @albanD

Differential Revision: [D75985423](https://our.internmc.facebook.com/intern/diff/D75985423)

larryliu0820 committed:
[BE] Cleanup old ExecuTorch codegen and runtime code

Summary: These files are added to pytorch/pytorch before ExecuTorch is
opensourced. Now is a good time to remove it from pytorch/pytorch, since
the code is moved to pytorch/executorch already.

Test Plan: Rely on CI jobs.

[ghstack-poisoned]

larryliu0820 committed:
Update on "[BE] Cleanup old ExecuTorch codegen and runtime code"


Summary: These files are added to pytorch/pytorch before ExecuTorch is
opensourced. Now is a good time to remove it from pytorch/pytorch, since
the code is moved to pytorch/executorch already.

Test Plan: Rely on CI jobs.

cc albanD

[ghstack-poisoned]

kimishpatel reviewed (approved):
What can go wrong when dr. CI is around. lol

larryliu0820 committed:
Update on "[BE] Cleanup old ExecuTorch codegen and runtime code"


Summary: These files are added to pytorch/pytorch before ExecuTorch is
opensourced. Now is a good time to remove it from pytorch/pytorch, since
the code is moved to pytorch/executorch already.

Test Plan: Rely on CI jobs.

cc albanD

[ghstack-poisoned]

larryliu0820 committed:
Update on "[BE] Cleanup old ExecuTorch codegen and runtime code"


Summary: These files are added to pytorch/pytorch before ExecuTorch is
opensourced. Now is a good time to remove it from pytorch/pytorch, since
the code is moved to pytorch/executorch already.

Test Plan: Rely on CI jobs.

cc albanD

[ghstack-poisoned]

larryliu0820 commented:
@pytorchbot merge

cyyever commented:
@pytorchbot merge -i

nautsimon commented:
seeing an issue with setup script potentially related to these changes on master, may need a follow up change to core_generated_declarations_yaml.cmake
```
python3 setup.py develop         
Building wheel torch-2.8.0a0+gitcd3d2b7
-- Building version 2.8.0a0+gitcd3d2b7
-- Checkout nccl release tag: v2.26.5-1
cmake --build . --target install --config Release -j 12
ninja: error: rebuilding 'build.ninja': '... pytorch/torchgen/executorch/__init__.py', needed by 'aten/src/ATen/core_generated_declarations_yaml.cmake', missing and no known rule to make it
```

larryliu0820 commented:
> seeing an issue with setup script potentially related to these changes on master, may need a follow up change to core_generated_declarations_yaml.cmake
> 
> ```
> python3 setup.py develop         
> Building wheel torch-2.8.0a0+gitcd3d2b7
> -- Building version 2.8.0a0+gitcd3d2b7
> -- Checkout nccl release tag: v2.26.5-1
> cmake --build . --target install --config Release -j 12
> ninja: error: rebuilding 'build.ninja': '... pytorch/torchgen/executorch/__init__.py', needed by 'aten/src/ATen/core_generated_declarations_yaml.cmake', missing and no known rule to make it
> ```

Hmm could it be a legacy cmake cache? Can you run cmake clean and retry?

But yeah I was hoping to fix the executorch job failure before merging this. I can send a quick follow up tomorrow.

seemethere commented:
@pytorchbot revert -c ghfirst -m "This is failing when attempting to test against executorch main internally, author has acknowledged that this should be reverted"

larryliu0820 committed:
Update on "[BE] Cleanup old ExecuTorch codegen and runtime code"


Summary: These files are added to pytorch/pytorch before ExecuTorch is
opensourced. Now is a good time to remove it from pytorch/pytorch, since
the code is moved to pytorch/executorch already.

Test Plan: Rely on CI jobs.

cc albanD

[ghstack-poisoned]

larryliu0820 committed:
Update on "[BE] Cleanup old ExecuTorch codegen and runtime code"


Summary: These files are added to pytorch/pytorch before ExecuTorch is
opensourced. Now is a good time to remove it from pytorch/pytorch, since
the code is moved to pytorch/executorch already.

Test Plan: Rely on CI jobs.

cc albanD

Differential Revision: [D75985423](https://our.internmc.facebook.com/intern/diff/D75985423)

[ghstack-poisoned]

larryliu0820 commented:
@pytorchbot merge -i “seems unrelated. The ET failure can be fixed in a follow up”
